grpc/src/main/java/io/stargate/grpc/service/interceptors/metrics/MetricCollectingServerInterceptor.java

```diff
@@ -0,0 +1,289 @@
+/*
```

Comment:
Had to pull this in directly or else there were issues with the classloader.

Reply:
Let me investigate a bit on this class loading, should not be hapenning..

Reply:
OK so here is the patch that solves it. Solution:

* load the `io.micrometer.core.instrument.binder.grpc` with the class loader of the `grpc` module (needed because this depends on the `io.grpc` classes)
* load the rest of the micrometer stuff by the `core` class loader (needed not to break the interaction with the MeterRegistry)

I know it looks strange, but this is one of the bizarre situations with the OSGi. Starup now works, did not test further, I have no idea how to simulate grpc traffic. Patch :arrow_down: , I did not include the clean-up.

```diff --git a/grpc/pom.xml b/grpc/pom.xml
index 7d807865..5ab328dc 100644
--- a/grpc/pom.xml
+++ b/grpc/pom.xml
@@ -52,6 +52,10 @@
       <version>${grpc.version}</version>
     </dependency>
     <!-- Other -->
+    <dependency>
+      <groupId>io.micrometer</groupId>
+      <artifactId>micrometer-core</artifactId>
+    </dependency>
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
@@ -116,11 +120,13 @@
                             org.apache.cassandra.stargate,
                             org.apache.cassandra.stargate.*,
                             org.javatuples,
-                            com.codahale.metrics,
                             com.datastax.oss.driver.api.core,
                             com.datastax.oss.driver.api.core.*,
                             com.datastax.oss.driver.shaded.guava.*,
-                            io.micrometer.*
+                            com.codahale.metrics,
+                            com.codahale.metrics.*,
+                            !io.micrometer.core.instrument.binder.grpc,
+                            io.micrometer.core.*
                         ]]></Import-Package>
             <Export-Package>!*</Export-Package>
             <Embed-Dependency>*;scope=compile|runtime</Embed-Dependency>
diff --git a/grpc/src/main/java/io/stargate/grpc/impl/GrpcImpl.java b/grpc/src/main/java/io/stargate/grpc/impl/GrpcImpl.java
index 0d32143d..6f94962c 100644
--- a/grpc/src/main/java/io/stargate/grpc/impl/GrpcImpl.java
+++ b/grpc/src/main/java/io/stargate/grpc/impl/GrpcImpl.java
@@ -17,13 +17,13 @@ package io.stargate.grpc.impl;
 
 import io.grpc.Server;
 import io.grpc.ServerBuilder;
+import io.micrometer.core.instrument.binder.grpc.MetricCollectingServerInterceptor;
 import io.stargate.auth.AuthenticationService;
 import io.stargate.core.metrics.api.Metrics;
 import io.stargate.db.Persistence;
 import io.stargate.grpc.service.Service;
 import io.stargate.grpc.service.interceptors.AuthenticationInterceptor;
 import io.stargate.grpc.service.interceptors.RemoteAddressInterceptor;
-import io.stargate.grpc.service.interceptors.metrics.MetricCollectingServerInterceptor;
 import java.io.IOException;
 import java.io.UncheckedIOException;
 import org.slf4j.Logger;
diff --git a/pom.xml b/pom.xml
index 748358c6..537587a7 100644
--- a/pom.xml
+++ b/pom.xml
@@ -29,8 +29,8 @@
     <slf4j.version>1.7.30</slf4j.version>
     <logback.version>1.2.3</logback.version>
     <!-- Make sure micrometer and prometheus versions are in sync  -->
-    <micrometer.version>1.6.5</micrometer.version>
-    <prometheus.version>0.9.0</prometheus.version>
+    <micrometer.version>1.7.0</micrometer.version>
+    <prometheus.version>0.10.0</prometheus.version>
     <junit.version>5.7.0</junit.version>
     <mockito.version>3.5.13</mockito.version>
     <jacoco.version>0.8.6</jacoco.version>
@@ -174,6 +174,7 @@
         <module>rate-limiting-global</module>
         <module>grpc-proto</module>
         <module>grpc</module>
+        <module>metrics-jersey</module>
       </modules>
     </profile>
     <profile>
/*
```

Reply:
Okay so this works for start up because the `grpc` classloader is loading `MetricCollectingServerInterceptor` but looks like I've got some more work to do because it fails when a request is made since the `core` classloader loads `MeterRegistry`.